go-controller/pkg/clustermanager/healthcheck/healthcheck.go

```diff
@@ -19,6 +23,38 @@ func healthStateFromError(err error) HealthState {
 		return AVAILABLE
 	}
 
+	if errors.Is(err, healthcheck.ErrNotServing) {
+		// connected fine but not serving
+		return UNAVAILABLE
```

Comment:
I have questions about how we are determining if a node is unavailable. It looks like we are looking at ErrNotServing, which means you got a GRPC response with that code. I'm not sure yet what that code actually means from the server side. I also see there are other non-positive codes like service_unknown and unknown...not sure how they are different yet.

Reply:
We were not using the other codes before and just expected `HealthCheckResponse_SERVING`.

Here I propose to interpret `HealthCheckResponse_SERVING` as AVAILABLE (as before), any other grpc response code and tcp connection refused as UNAVAILABLE (new in this PR) , and any other outcome as UNREACHABLE.